tests/test_gpu_examples.py

```diff
@@ -1435,6 +1438,195 @@ def test_offload_merge(self):
+
+        # logits from quantized LoRA model using PiSSA
+        lora_config = LoraConfig(
+            task_type=task_type,
+            init_lora_weights="pissa",
+            target_modules=target_modules,
+        )
+        pissa_model = cls.from_pretrained(model_id).eval().to(device)
+        pissa_model = get_peft_model(pissa_model, lora_config)
+
+        # save LoRA weights, they should be initialized such that they minimize the quantization error
+        pissa_model.base_model.peft_config["default"].init_lora_weights = True
+        pissa_model.save_pretrained(tmp_path / "pissa_model")
+
+        pissa_model = pissa_model.unload()
+        pissa_model.save_pretrained(tmp_path / "residual_model")
+
+        del pissa_model
+        gc.collect()
+        torch.cuda.empty_cache()
+
+        # now load quantized model and apply PiSSA-initialized weights on top
+        qpissa_model = self.quantize_model(
+            cls.from_pretrained(tmp_path / "residual_model").eval().to(device), bits, device
+        )
+        qpissa_model = PeftModel.from_pretrained(qpissa_model, tmp_path / "pissa_model")
+        qpissa_model = qpissa_model.merge_and_unload()
+        qpissa_error = self.nuclear_norm(base_model, qpissa_model)
+        print(qpissa_error)
+        del qpissa_model
+        gc.collect()
+        torch.cuda.empty_cache()
+
+        assert qlora_error > 0.0
+        assert qpissa_error > 0.0
+
+        # next, check that PiSSA quantization errors are smaller than LoRA errors by a certain margin
+        assert qpissa_error < (qlora_error / self.error_factor)
+
+    @pytest.mark.parametrize("device", ["cuda", "cpu"])
+    def test_bloomz_pissa_4bit(self, device, tmp_path):
+        # In this test, we compare the logits of the base model, the quantized LoRA model, and the quantized model
+        # using PiSSA. When quantizing, we expect a certain level of error. However, we expect the PiSSA quantized
+        # model to have less error than the normal LoRA quantized model. Note that when using normal LoRA, the
+        # quantization error is simply the error from quantization without LoRA, as LoRA is a no-op before training.
+        # We still apply LoRA for the test for consistency.
+
+        self.get_errors(bits=4, device=device, tmp_path=tmp_path)
+
+    @pytest.mark.parametrize("device", ["cuda", "cpu"])
+    def test_bloomz_pissa_8bit(self, device, tmp_path):
+        # Same test as test_bloomz_pissa_4bit but with 8 bits.
+        self.get_errors(bits=8, device=device, tmp_path=tmp_path)
+
+    @pytest.mark.parametrize("device", ["cuda", "cpu"])
+    def test_t5_pissa_4bit(self, device, tmp_path):
+        self.get_errors(bits=4, device=device, model_id="t5-small", tmp_path=tmp_path)
+
+    @pytest.mark.parametrize("device", ["cuda", "cpu"])
+    def test_t5_pissa_8bit(self, device, tmp_path):
+        self.get_errors(bits=8, device=device, model_id="t5-small", tmp_path=tmp_path)
+
+
+@pytest.mark.xfail(
```

Comment:
This whole test needs to be indented one level to be within the class. Then you can remove the `skipif` decorator below, since the whole class already has it.